orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -165,8 +165,20 @@ class CompleteStageHandler(
       }
 
       val notStartedSynthetics = execution.stages.filter { it.parentStageId == id && it.status == NOT_STARTED}
+      val notStartedSyntheticRefIds = notStartedSynthetics.map { it.refId }
 
       builder.buildAfterStages(this, onFailureStages) { it: Stage ->
+        if (it.requisiteStageRefIds.any { notStartedSyntheticRefIds.contains(it) }) {
+          /*
+           * Avoid having any NOT_STARTED synthetic stage as a requisiteStageRefId as they will be subsequently removed
+           * from the execution.
+           *
+           * Opting for an emptySet() as the StartStageHandler will reject any attempt to start a stage with an
+           * upstream failure.
+           */
+          it.requisiteStageRefIds = emptySet()
```

Comment:
Alternatively I suppose `StartStage` message could be altered to include a flag indicating that upstream failures should be ignored.

This _seemed_ simpler and fit with the way that `Stage.ancestors()` is working. 

It fixes a problem wherein an upstream stage output was not being found because it was attempting to navigate using a `requisiteStageRefId` that did not exist versus using `parentStageId`. 

Thoughts? I'm not against modifying the `StartStageHandler` or equivalent if we think that's cleaner.

Reply:
Another alternative would have been to have `Stage.ancestors()` filter out any `requisiteStageRefId` that reference something that no longer exists.

Reply:
It seems like filtering it would make sense. What if there are multiple requisite stage ids some of which are SUCCEEDED? You'd still remove all of them? I guess it makes no practical difference but it seems like we lose information.

Reply:
That's fair ... I did consider it initially and will flip back to filtering. Right now we don't have synthetic stages with multiple requisites (afaik) but that might not always be the case.